sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -356,21 +379,25 @@ DataSource buildDatasource() throws Exception {
         current = basicDataSource;
       }
 
-      // wrapping the datasource as a pooling datasource
-      DataSourceConnectionFactory connectionFactory = new DataSourceConnectionFactory(current);
-      PoolableConnectionFactory poolableConnectionFactory =
-          new PoolableConnectionFactory(connectionFactory, null);
-      GenericObjectPoolConfig poolConfig = new GenericObjectPoolConfig();
-      poolConfig.setMaxTotal(1);
-      poolConfig.setMinIdle(0);
-      poolConfig.setMinEvictableIdleTimeMillis(10000);
-      poolConfig.setSoftMinEvictableIdleTimeMillis(30000);
-      GenericObjectPool connectionPool =
-          new GenericObjectPool(poolableConnectionFactory, poolConfig);
-      poolableConnectionFactory.setPool(connectionPool);
-      poolableConnectionFactory.setDefaultAutoCommit(false);
-      poolableConnectionFactory.setDefaultReadOnly(false);
-      return new PoolingDataSource(connectionPool);
+      if (getWrapAsPoolDataSource().get()) {
```

Comment:
Can we just check here if provided data source (`current`) is instance of `PoolingDataSource`? In this case we could detect it automatically and no need to add another `with*` option.

Reply:
I won't be enough. Some DataSource are poolable but not instance of DBCP `PoolingDataSource`. For instance `OracleDataSource` is poolable, not related to DBCP. The user can also provide C3P0 DataSource.

Reply:
Ok, then another question - do we need to have `withWrapAsPoolableDataSource` as a dedicated method? I think we can just overload `DataSourceConfiguration.create()` and add a boolean argument `isPoolingDataSource` for this case and, by default, `isPoolingDataSource = true`.

Reply:
Agree, updated.